core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.controller.ts

```diff
@@ -215,20 +225,102 @@ angular.module('oppia').directive('topicsAndSkillsDashboardPage', [
+          ctrl.changePage = function(str) {
+            ctrl.lastPage = parseInt(
+              String(ctrl.totalCount / ctrl.itemsPerPage));
+            if (str === ctrl.PREV_PAGE && ctrl.pageNumber >= 1) {
+              ctrl.paginationHandler(ctrl.pageNumber - 1);
+            } else if (str === ctrl.NEXT_PAGE &&
+                ctrl.pageNumber < ctrl.lastPage) {
+              ctrl.paginationHandler(ctrl.pageNumber + 1);
+            }
+          };
+          ctrl.applyFilters = function() {
+            ctrl.topicSummaries = (
+              TopicsAndSkillsDashboardPageService.getFilteredTopics(
+                ctrl.totalTopicSummaries, ctrl.filterOptions));
+            ctrl.displayedTopicSummaries =
+                ctrl.topicSummaries.slice(0, (1) * ctrl.itemsPerPage);
+            ctrl.currentCount = ctrl.topicSummaries.length;
+            ctrl.pageNumber = 0;
+          };
+          ctrl.resetFilters = function() {
+            ctrl.topicSummaries = ctrl.totalTopicSummaries;
+            ctrl.currentCount = ctrl.totalCount;
+            ctrl.filterOptions = {
+              sort: '',
+              keywords: '',
+              category: '',
+              status: '',
+            };
+            ctrl.paginationHandler(0);
+          };
+          ctrl.itemsPerPageHandler = function() {
+            ctrl.paginationHandler(0);
+          };
+          ctrl.createTopicSkill = function() {
+            if (ctrl.activeTab === ctrl.TAB_NAME_TOPICS) {
+              ctrl.createTopic();
+            } else if (ctrl.activeTab === ctrl.TAB_NAME_UNTRIAGED_SKILLS) {
+              ctrl.createSkill();
+            }
+          };
+
           ctrl.$onInit = function() {
             ctrl.TAB_NAME_TOPICS = 'topics';
+            ctrl.NEXT_PAGE = 'next_page';
+            ctrl.PREV_PAGE = 'prev_page';
             ctrl.TAB_NAME_UNTRIAGED_SKILLS = 'untriagedSkills';
             ctrl.TAB_NAME_UNPUBLISHED_SKILLS = 'unpublishedSkills';
+            ctrl.pageNumber = 0;
+            ctrl.topicPageNumber = 0;
+            ctrl.itemsPerPage = 10;
+            ctrl.skillPageNumber = 0;
+            ctrl.selectedIndex = null;
+            ctrl.itemsPerPageChoice = [10, 15, 20];
+            ctrl.filterOptions = {
+              sort: '',
+              keywords: '',
+              category: '',
+              status: '',
+            };
+            ctrl.categories = TOPIC_CATEGORIES;
+            ctrl.sortOptions = (E_SORT_OPTIONS);
+            ctrl.statusOptions = (E_PUBLISHED_OPTIONS);
+
+            ctrl.repeater = function(range) {
+              var arr = [];
+              for (var i = 0; i < range; i++) {
+                arr.push(i);
+              }
+              return arr;
+            };
             $scope.$on(
               EVENT_TOPICS_AND_SKILLS_DASHBOARD_REINITIALIZED, function(
                   evt, stayInSameTab) {
-                _initDashboard(stayInSameTab);
+                ctrl._initDashboard(stayInSameTab);
```

Comment:
If initDashboard is not being accessed "from outside" then do not bind it to ctrl. Use ctrl only for things which need to be accessible to the corresponding template.

Reply:
Needed to test it. (It's not being called from outside.)

Reply:
/cc @marianazangrossi to confirm whether this is OK.